### PR TITLE
fix(appbuilder): SAM local debugging: region and credential in launch config should override toolkit default

### LINKED ISSUE #5957: SAM regression: v3.31.0 does not use `aws.credentials` profile for local debugging
## Problem
Using aws-sam debug profile to test a lambda locally.  As of v3.31.0, the selected profile no longer is getting correctly passed down to our go libraries

## Steps to reproduce the issue

When using a VSCode debug configuration like so...
```
{
    "name": "PROD invoice upload",
    "type": "aws-sam",
    "request": "direct-invoke",
    "invokeTarget": {
        "target": "code",
        "lambdaHandler": "lambda",
        "projectRoot": "/some/src/code"
    },
				
    "aws": {
        "region": "us-west-2",
	"credentials": "profile:dev",          <--------------- profile selected
    },
    "sam": {
        "localArguments": ["--debugger-path=/home/vagrant/go/bin"],
	"buildArguments": []
    },
    "lambda": {
        "runtime": "go1.x",
	"environmentVariables": { 
    ....
```

The following go code no longer behaves as expected, instead getting an empty profile and defaulting everything to "us-east1"

```
"github.com/aws/aws-sdk-go-v2/config"
... 
    // Create aws configuration...
    awsCfg, err := config.LoadDefaultConfig(ctx)
    if err != nil {
	    return err
    }

    // Then instantiate an S3 client object using the aws configuration
    // (not to be confused with the cfg parameter passed into the Handler)
    stsClient := sts.NewFromConfig(awsCfg)
...
```

## Expected behavior

LoadDefaultConfig() uses the selected SSO profile from the VSCode debug config.  Not 100% sure what the critical difference is... presumably some environmental variables are no longer being correctly passed through to SAM?

## System details (run `AWS: About` and/or `Amazon Q: About`)

-   OS: linux
-   Visual Studio Code version: 1.93.0
-   AWS Toolkit version:  3.31.0 (but NOT v3.29.0)
-   Amazon Q version: irrelevant

### PR DESCRIPTION
**Pull request by roger-zhangg**
## Problem
fix #5957 
Appbuilder added `--region` parameter to `sam local invoke`, however this region is always reading region prop from default toolkit connection. Ignoring region set in sam debug config.

## Solution
region and credential set in sam debug config should override region and credential in current active toolkit connection.

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/toolkit/.changes/next-release/Bug Fix-e5a76bd7-06a7-4a59-aead-8ea3e3924ee9.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Bug Fix",
+	"description": "region in debug config should override default toolkit region"
```
Comment (justinmk3) [inline packages/toolkit/.changes/next-release/Bug Fix-e5a76bd7-06a7-4a59-aead-8ea3e3924ee9.json:3]: always helpful to customers to mention exact field names. also, markdown is supported.

also, https://github.com/aws/aws-toolkit-vscode/pull/6011 is about `aws.credentials` , not regions? If there are multiple issues being fixed, it's definitely worth creating multiple changelog items.

```suggestion
	"description": "SAM local debugging: `aws.region` specified in launch config should override default Toolkit region"
```

Comment (roger-zhangg) [reply packages/toolkit/.changes/next-release/Bug Fix-e5a76bd7-06a7-4a59-aead-8ea3e3924ee9.json:3]: Sure, updating

File: packages/core/src/test/shared/sam/debugger/samDebugConfigProvider.test.ts [hunk 2]
```diff
@@ -2914,6 +2914,7 @@ describe('SamDebugConfigurationProvider', async function () {
             const tempDir = path.dirname(actual.codeRoot)
 
             const expected: SamLaunchRequestArgs = {
+                // the credential in debug config will override default toolkit credential
```
Comment (justinmk3) [inline packages/core/src/test/shared/sam/debugger/samDebugConfigProvider.test.ts:2917]:
```suggestion
                // The `aws.credentials` field in debug config, overrides default toolkit credentials.
```

File: packages/core/src/test/shared/sam/debugger/samDebugConfigProvider.test.ts [hunk 3]
```diff
@@ -2949,7 +2950,8 @@ describe('SamDebugConfigurationProvider', async function () {
                 templatePath: pathutil.normalize(path.join(actual.baseBuildDir!, 'app___vsctk___template.yaml')),
                 parameterOverrides: undefined,
                 architecture: 'x86_64',
-                region: 'us-west-2',
+                // the region in debug config should override default toolkit region
```
Comment (justinmk3) [inline packages/core/src/test/shared/sam/debugger/samDebugConfigProvider.test.ts:2953]:
```suggestion
                // The `aws.region` field in debug config, overrides default toolkit region.
```

**Pull request merged**